modules/programs/ne.nix

```
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.programs.ne;
+
+  autoPrefFiles = let
+    autoprefs = if cfg.defaultPreferences != "" then
+      cfg.automaticPreferences // {
+        ".default" = cfg.defaultPreferences;
+      } # include defaultPreferences as .default#ap
+    else
+      cfg.automaticPreferences;
+    keys = builtins.attrNames autoprefs;
+    gen = key: {
+      name = ".ne/${key}#ap"; # file path
+      value = {
+        text = builtins.getAttr key autoprefs;
+      }; # content, in the expected "path".text format
+    };
+  in listToAttrs (map gen keys); # assembling these into a regular record
```

Could use [`mapAttrs'`](https://nixos.org/nixpkgs/manual/#function-library-lib.attrsets.mapAttrs-prime). Something like
```suggestion
  in mapAttrs' gen autoprefs;
```
where `gen` is adjusted to take both the attribute name and value.